File: GameLibrary/src/cc/lib/swing/AWTRotationImageFilter.java
package cc.lib.swing;

import java.awt.image.RGBImageFilter;

class AWTRotationImageFilter extends RGBImageFilter {

    final int [] source;
    final int degrees;
    final int srcWid, srcHgt, dstWid, dstHgt;
    
    
    public AWTRotationImageFilter(int[] source, int degrees, int srcWid, int srcHgt, int dstWid, int dstHgt) {
        super();
        this.source = source;
        this.degrees = degrees;
        this.srcWid = srcWid;
        this.srcHgt = srcHgt;
        this.dstWid = dstWid;
        this.dstHgt = dstHgt;
    }


    @Override
    public int filterRGB(int x, int y, int rgb) {

        int sx = 0;
        int sy = 0;
        switch (degrees) {
            case 0:break;
            case 90:
                sx = y; sy = x; break;
            case 180:
                sx = srcWid - x; 
                sy = srcHgt - y; break;
            case 270:
                sx = srcHgt - y; 
                sy = srcWid - x; break;
        }
        
        return source [sx + sy * srcWid];
    }

}

File: SwingApps/src/cc/lib/swing/AWTRotationImageFilter.kt
package cc.lib.swing

import java.awt.image.RGBImageFilter

internal class AWTRotationImageFilter(val source: IntArray, val degrees: Int, val srcWid: Int, val srcHgt: Int, val dstWid: Int, val dstHgt: Int) : RGBImageFilter() {
	override fun filterRGB(x: Int, y: Int, rgb: Int): Int {
		var sx = 0
		var sy = 0
		when (degrees) {
			0 -> {}
			90 -> {
				sx = y
				sy = x
			}
			180 -> {
				sx = srcWid - x
				sy = srcHgt - y
			}
			270 -> {
				sx = srcHgt - y
				sy = srcWid - x
			}
		}
		return source[sx + sy * srcWid]
	}
}